subsys/mgmt/mcumgr/smp/src/smp.c

```diff
@@ -318,24 +321,37 @@ int smp_process_request_packet(struct smp_streamer *streamer, void *vreq)
 			break;
 		}
 
-		rsp = smp_alloc_rsp(req, streamer->smpt);
-		if (rsp == NULL) {
-			rc = MGMT_ERR_ENOMEM;
-			break;
-		}
-
-		cbor_nb_reader_init(streamer->reader, req);
-		cbor_nb_writer_init(streamer->writer, rsp);
-
-		/* Process the request payload and build the response. */
-		rc = smp_handle_single_req(streamer, &req_hdr, &handler_found, &rsn);
-		if (rc != 0) {
-			break;
+		if (req_hdr.nh_op == MGMT_OP_READ || req_hdr.nh_op == MGMT_OP_WRITE) {
+			cbor_nb_reader_init(streamer->reader, req);
+			rsp = smp_alloc_rsp(req, streamer->smpt);
+			if (rsp == NULL) {
+				rc = MGMT_ERR_ENOMEM;
+				break;
+			}
+
+			cbor_nb_writer_init(streamer->writer, rsp);
+			/* Process the request payload and build the response. */
+			rc = smp_handle_single_req(streamer, &req_hdr, &handler_found, &rsn);
+			if (rc != 0) {
+				break;
+			}
+
+			/* Send the response. */
+			rc = streamer->smpt->output(rsp);
+			rsp = NULL;
+		} else if (req_hdr.nh_op == MGMT_OP_READ_RSP ||
```

Comment:
You can add `IS_ENABLED(CONFIG_SMP_CLIENT) && ` to if and in such case, you can avoid #ifdef below.